api/v1alpha1/mariadb_types.go

```diff
@@ -238,10 +238,14 @@ type MariaDBSpec struct {
 	// +kubebuilder:default=3306
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number","urn:alm:descriptor:com.tectonic.ui:advanced"}
 	Port int32 `json:"port,omitempty"`
+	// EphemeralStorage indicates whether to use ephemeral storage for the instances.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	EphemeralStorage *bool `json:"ephemeralStorage,omitempty" webhook:"inmutable"`
 	// VolumeClaimTemplate provides a template to define the Pod PVCs.
-	// +kubebuilder:validation:Required
+	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	VolumeClaimTemplate VolumeClaimTemplate `json:"volumeClaimTemplate" webhook:"inmutable"`
+	VolumeClaimTemplate *VolumeClaimTemplate `json:"volumeClaimTemplate" webhook:"inmutable"`
```

Comment:
You can remove the pointer here.

Either we use ephemeral or regular storage, the struct will be filled at some point, so allowing the field to be nil does not worth it IMHO. This way we also reduce the risk of a nil pointer exception.